sched/definitions.go

```diff
@@ -17,3 +17,33 @@ type Task struct {
 	Command    []string // Argv to run
 	SnapshotId string
 }
+
+type JobState int
+
+const (
+	// An unambiguous 0-value.
+	UNKNOWN JobState = iota
+	// Waiting to run.
+	PENDING
+	// Running
+	RUNNING
+	// Done
+	DONE
+)
+
+type JobStatus struct {
```

Comment:
If the goal is to only allow folks to use the factory methods, should JobStatus be internal (jobStatus)